src/main.cpp

```diff
@@ -5911,10 +6135,12 @@ bool SendMessages(CNode* pto)
             NodeId staller = -1;
             FindNextBlocksToDownload(pto->GetId(), MAX_BLOCKS_IN_TRANSIT_PER_PEER - state.nBlocksInFlight, vToDownload, staller);
             BOOST_FOREACH(CBlockIndex *pindex, vToDownload) {
-                vGetData.push_back(CInv(MSG_BLOCK, pindex->GetBlockHash()));
-                MarkBlockAsInFlight(pto->GetId(), pindex->GetBlockHash(), consensusParams, pindex);
-                LogPrint("net", "Requesting block %s (%d) peer=%d\n", pindex->GetBlockHash().ToString(),
-                    pindex->nHeight, pto->id);
+                if (State(pto->GetId())->fHaveWitness || !IsWitnessEnabled(pindex, consensusParams)) {
```

Comment:
If I am understanding this right, then I think we wouldn't download (via parallel-fetch at least) the very last block before segwit is activated from a non-NODE_WITNESS peer?  I can't imagine this being a significant issue but I thought it might be clearer anyway to make this `!IsWitnessEnabled(pindex->pprev, )`.